jib-gradle-plugin/src/test/java/com/google/cloud/tools/jib/gradle/GradleProjectPropertiesExtensionTest.java

```diff
@@ -101,23 +117,83 @@ private FooExtensionConfig(Map<String, String> properties) {
     public String getExtensionClass() {
       return extensionClass;
     }
+
+    @Override
+    public Optional<Object> getExtraConfiguration() {
+      return Optional.ofNullable(extraConfig);
+    }
+  }
+
+  private static class FooExtensionConfig extends BaseExtensionConfig<ExtensionDefinedFooConfig> {
+
+    private FooExtensionConfig() {
+      super(FooExtension.class.getName(), Collections.emptyMap(), null);
+    }
+
+    private FooExtensionConfig(Map<String, String> properties) {
+      super(FooExtension.class.getName(), properties, null);
+    }
+
+    private FooExtensionConfig(ExtensionDefinedFooConfig extraConfig) {
+      super(
+          FooExtension.class.getName(),
+          Collections.emptyMap(),
+          new Action<ExtensionDefinedFooConfig>() {
+            @Override
+            public void execute(ExtensionDefinedFooConfig instance) {
+              instance.potato = extraConfig.potato;
+            }
+          });
+    }
   }
 
-  private static class BarExtensionConfig extends FooExtensionConfig {
+  private static class BarExtensionConfig extends BaseExtensionConfig<ExtensionDefinedBarConfig> {
 
     private BarExtensionConfig() {
-      super(BarExtension.class.getName());
+      super(BarExtension.class.getName(), Collections.emptyMap(), null);
+    }
+
+    private BarExtensionConfig(ExtensionDefinedBarConfig extraConfig) {
+      super(
+          BarExtension.class.getName(),
+          Collections.emptyMap(),
+          new Action<ExtensionDefinedBarConfig>() {
+            @Override
+            public void execute(ExtensionDefinedBarConfig instance) {
+              instance.tomato = extraConfig.tomato;
+            }
+          });
+    }
+  }
+
+  // Not to be confused with Jib's plugin extension config. This class is for an extension-defined
+  // config specific to a third-party extension.
+  private static class ExtensionDefinedFooConfig {
+
+    private String potato;
```

Comment:
as much as I love potato and tomato(and I really really do), maybe these should be more descriptive?
`fooParam` `barParam`